modules/calc/src/main/java/com/opengamma/strata/calc/runner/CalculationParameter.java

```diff
@@ -23,7 +35,7 @@
    * 
    * @return the type of the parameter implementation
    */
-  public abstract Class<?> queryType();
+  public abstract Class<? extends CalculationParameter> queryType();
```

Comment:
Should the return type be the implementing type? If that's the case we should have a type parameter to make that happen.

With this signature it's perfectly possible for a parameter implementation to return a totally unrelated parameter type from `queryType`.